src/warc2zim/content_rewriting/css.py

```diff
@@ -23,14 +36,26 @@ def rewrite(self, content: str | bytes) -> str:
             rules = parse_stylesheet(content)
         self.process_list(rules)
 
-        output = serialize(rules)
+        try:
+            output = serialize(rules)
+        except Exception:
+            fallback_rewriter = FallbackRegexCssRewriter(self.url_rewriter)
+            if isinstance(content, bytes):
+                content = content.decode()
+            return fallback_rewriter.rewrite_content(
+                content, {}  # pyright: ignore[reportArgumentType]
```

Comment:
Why do we need this pyright rule? I checked but I don't understand why we have an error and need this rule (maybe add a comment to explain it so that it is clearer for next maintainer)

Reply:
This is because pyright "promotes" `bytes` type to `bytes|memoryview`.
See the first commit.

Maybe it could be backported in our hatch config.

Reply:
Great! I approve your solution, see https://github.com/openzim/_python-bootstrap/issues/37